src/hotspot/os/linux/cgroupSubsystem_linux.cpp

```diff
@@ -308,18 +335,21 @@ bool CgroupSubsystemFactory::determine_type(CgroupInfo* cg_infos,
           cg_infos[CPUSET_IDX]._data_complete = true;
```

Comment:
Because `cpusets` might be mounted to `/cpusets` or `/dev/cpusets` and we don't know if this manual mount point comes before or after any of the others which belong to a Cgroup. If the manual mount point comes first in `mountinfo`, `check_mount_path()` would record its mount path as the *correct* one and would bail out later, when checking the other controllers.

The solution you propose in [jerboaa@1323315](https://github.com/jerboaa/jdk/commit/1323315036c0b7625cd1a05690b5944df8457bad) is of course simpler, but it assumes that the Cgroup controllers are always mounted under `/sys/fs/cgroup`. I'm not sure about that and I just wanted to avoid the next issue when popping up when the controllers are mounted to a different location. But if you say they are always mounted under `/sys/fs/cgroup` I'm happy to use the simpler solution.

Reply:
> If the manual mount point comes first in mountinfo, check_mount_path() would record its mount path as the correct one and would bail out later, when checking the other controllers.

Would it? In that case we'd have `mount_path == NULL` and then the path gets checked against `/sys/fs/cgroup`, which it isn't and gets skipped. If what you said was true, wouldn't [this patch](https://github.com/jerboaa/jdk/commit/97bb674c6fa9031d5ca34d3de74bba16c5afdb46) trigger the assertion in the else clause in `check_mount_path()` when running the new regression test, which has data that you describe?